ldelf/ta_elf_rel.c

```diff
@@ -312,28 +329,31 @@ static void e64_relocate(struct ta_elf *elf, unsigned int rel_sidx)
 	}
 
 	/* Check the address is inside TA memory */
-	assert(shdr[rel_sidx].sh_addr < (elf->max_addr - elf->load_addr));
+	if (ADD_OVERFLOW(shdr[rel_sidx].sh_addr,
```

Comment:
same comment about `shdr[rel_sidx].sh_addr` maybe not checked against max valid address (actually offset).

Reply:
Same answer for `shdr[rel_sidx].sh_addr` :-)